### PR TITLE
feat(engine): implement ADR-0036 D-REASSESS post-AM reassessment and claim gate (issue #436)

### PR DESCRIPTION
**Pull request by Szczepanov**
## Summary

Implements ADR-0036 D-REASSESS (post-AM reassessment before PM launch) and completes the analysis deliverable for [issue #436](https://github.com/Szczepanov/adaptive-training-recommender/issues/436).

- **Pure reassessment engine (`app/src/engine/intradayReassessment.ts`):**
  - Implements `reassessDependentBundleMember` and `computeReassessmentInputRevision`.
  - Evaluates predecessor completion state and completion timestamp.
  - Verifies minimum timing separation (ADR-0036 D-TIME) using `elapsedMinutesBetweenInstants`.
  - Verifies explicit post-predecessor symptom and response confirmation (`SessionResponse` and `RegionTissueResponse`) per ADR-0023 D-MRESP; does not infer well-tolerated status from silence.
  - Detects acute reactive symptoms (severe tissue pain or adverse reactions) and scales or rejects PM work accordingly.
  - Enforces that favorable predecessor responses never expand authored dose beyond plan limits.
  - Verifies candidate admission against remaining daily capacity via `admitsCandidate` (`app/src/engine/dailyLedger.ts`).
  - Evaluates readiness & safety envelopes bypassing single-session `alreadyTrainedOverride` circuit breaker via `EvaluateReadinessOptions.ignoreAlreadyTrainedOverride`.
- **Readiness rules adjustment (`app/src/engine/rules.ts`):**
  - Adds `EvaluateReadinessOptions { ignoreAlreadyTrainedOverride?: boolean }` to `evaluateReadinessAndSafetyEnvelope` and `evaluateEnvelopes`.
  - Avoids erroneously clamping intraday double sessions to `recover` mode and `Rest` tier when same-day predecessor work has already been accounted for via ledger debits.
- **Atomic claim transaction (`app/src/services/sessionOccurrenceService.ts`):**
  - Implements `claimOccurrenceLaunch(userId, occurrenceId, now)` using Firestore `runTransaction`.
  - Atomically verifies `state === 'scheduled'` and transitions to `'active'`, preventing double-launches or race conditions.
- **Policy bump (`app/src/engine/policy.ts`):**
  - Advances `POLICY_VERSION` to `'2026-09-h4-intraday-reassessment-v1'` and archives previous policy to `HISTORICAL_POLICY_VERSIONS`.

## Risk and reviewer guidance

- **Low risk to existing single-session recommendation flows:** The `ignoreAlreadyTrainedOverride` option defaults to undefined/false. Existing morning recommendation flows (`evaluateTraining`, `evaluateTrainingWithIntent`, etc.) retain the single-session-per-day circuit breaker unchanged.
- **Strict safety fail-closed semantics:** If a predecessor session was not completed, or post-predecessor confirmation is absent, reassessment evaluates to `pending`, preventing premature PM launch. If severe/adverse tissue responses occur, it evaluates to `reject` with `safety_guardrail`.
- **Zero schema changes needed:** Directly reuses canonical `SessionResponse` (ADR-0023 D-MRESP) and `RegionTissueResponse`.

## Domain invariants

- **ADR-0036 D-REASSESS:** Runtime reassessment is deterministic and pure: given the same inputs and input revisions, `reassessDependentBundleMember` returns identical verdicts.
- **ADR-0036 D-TIME:** Minimum separation between prospective start instant and predecessor actual end instant is enforced using `elapsedMinutesBetweenInstants`.
- **ADR-0023 D-MRESP:** Explicit post-predecessor confirmation is required; absent confirmation leaves the decision `pending`.
- **ADR-0036 D-LEDGER:** Ledger capacity is checked prior to launching any secondary session.
- **No dose expansion:** Favorable responses cannot scale volume or intensity above authored limits.

## Validation

- **Unit tests:**
  - `app/src/engine/intradayReassessment.test.ts`: 10 comprehensive tests covering all edge cases (uncompleted predecessor, separation timing, missing confirmations, severe pain rejection, unexpected fatigue scaling, already-trained override bypass, ledger exhaustion).
  - `app/src/services/sessionOccurrenceService.test.ts`: 4 tests for atomic `claimOccurrenceLaunch` (successful claim, missing occurrence, non-scheduled state, corrupted document).
  - All 3,719 unit tests in `app` pass cleanly (`vitest run`), including all external-plan execution-binding tests from merged #440.
- **Engine verification:**
  - `npm run check` (typecheck + lint + tests + sports knowledge + workout catalog) passed with 0 errors.
  - `node scripts/check-policy-drift.mjs origin/main` verified policy version bump.
  - Pre-commit and pre-push hooks (formatting, linting, secrets, Warsaw date rule) passed cleanly.

## Screenshots or recordings

N/A — Backend and engine decision logic; no UI changes in this PR.

## Summary by CodeRabbit

- **New Features**
  - Sessions can be atomically claimed at launch, transitioning valid scheduled sessions to active.
  - Intraday reassessment considers tissue response, fatigue, timing, predecessor confirmation, readiness, safety, and available capacity.
  - Unexpected fatigue or moderate tissue irritation reduces session volume to 70% while preserving other dose settings.
  - Bundled intraday sessions can proceed without triggering the already-trained recovery override, while safety warnings remain enforced.
- **Bug Fixes**
  - Invalid or incomplete predecessor states, insufficient timing, missing confirmation, or severe tissue responses prevent dependent sessions from proceeding.
  - Invalid launch attempts leave session records unchanged.
- **Documentation**
  - Added planning documentation for reassessment rules, launch sequencing, validation, and revision tracking.

### CODE REVIEW

File: docs/plans/h4-dreassess-analysis.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+- **Date-level concurrency locking:**
+  ADR-0036 requires preventing two tabs or concurrent requests from spending the same minute
+  or systemic-cost capacity. Reading and updating only `session_occurrences/{occurrenceId}`
+  prevents double-launching the *same* session, but does not prevent two *different* sessions
+  from concurrently claiming the same remaining daily ledger capacity. The claim transaction
+  must read and write a **date-level document** (e.g. `users/{userId}/daily_recommendations/{date}`
+  or a dedicated daily ledger reservation record) holding active reservations and ledger revision.
+- **Reassessment input revision tracking:**
+  A composite revision/hash spanning availability, completed facts, check-in, and ledger
+  entries to verify the decision has not become stale at launch time.
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/plans/h4-dreassess-analysis.md:141]: Addressed: ReassessmentInputRevision has been updated in both the analysis document and the engine implementation (\pp/src/engine/intradayReassessment.ts\) with \postPredecessorConfirmationRevision?: string\ to version predecessor SessionResponse and tissue response confirmation records.

File: docs/plans/h4-dreassess-analysis.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+   ```
+   Generated when the provisional PM recommendation is composed, passed to the launch action,
+   and re-verified inside the atomic claim transaction; a mismatch means "stale, must recompute."
+
+2. **`reassessDependentBundleMember(...)` (pure engine function):**
+   - **Predecessor completion:** Verifies predecessor occurrence is `completed`. If not,
+     returns `pending`.
+   - **Elapsed separation:** If `minimumSeparationMinutes` is defined, computes
+     `elapsedMinutesBetweenInstants(evaluationInstant, predecessorCompletedAt)` via `localInstant.ts`.
+     If timestamps are missing or elapsed interval is insufficient, returns `pending` (unresolved timing prerequisite).
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/plans/h4-dreassess-analysis.md:173]: Addressed: Updated the separation calculation specification in \docs/plans/h4-dreassess-analysis.md\ to pass \prospectiveStartInstant\ as the first argument and \predecessorActualEndInstant\ as the second argument to \elapsedMinutesBetweenInstants\, matching the pure implementation in \pp/src/engine/intradayReassessment.ts\.

File: docs/plans/h4-dreassess-analysis.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+     and verifies `admitsCandidate` has headroom for both minutes and systemic cost.
+   - **Authored adjudication:** Re-runs `adjudicateAuthoredSession` against the current envelope.
+
+3. **Atomic launch claim transaction:**
+   Inside `runTransaction`:
+   - Reads the date-level reservation/recommendation document and the target `session_occurrence`.
+   - Verifies target occurrence is currently `scheduled` (rejects if already `active` or `completed`).
+   - Verifies date ledger revision matches the assumed revision (rejects with "stale, recompute" if mismatched).
+   - Transitions occurrence to `active`, registers the active reservation in the date ledger,
+     and commits atomically.
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/plans/h4-dreassess-analysis.md:190]: Addressed: Updated \docs/plans/h4-dreassess-analysis.md\ to specify reading current reservation/capacity state, re-verifying the full \ReassessmentInputRevision\, and atomically applying capacity consumption alongside occurrence transition in the claim transaction.

File: docs/plans/h4-dreassess-analysis.md [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+   - `POLICY_VERSION` bump.
+5. **D-AUDIT (issue #437):** Dedicated persistence shape for multi-decision same-day replay,
+   historical supersession chains, and replay verification suite.
+
+## Open questions for the implementing agent
+
+- Exact document path for date-level ledger locking: whether to lock on `users/{userId}/daily_recommendations/{date}`
+  or a dedicated `users/{userId}/daily_ledgers/{date}` document during the claim transaction.
+- Wiring `SessionResponse` into the immediate post-workout modal for external-plan sessions
+  so athletes can submit post-predecessor confirmation immediately upon completing the AM session.
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/plans/h4-dreassess-analysis.md]: Addressed: Updated \docs/plans/h4-dreassess-analysis.md\ under revised sequencing step 3 (\#434 PR 3\) to explicitly include wiring \SessionResponse\ and \	issueResponses\ prompt into the external-plan post-workout modal upon AM session completion, ensuring post-predecessor confirmation capture is a direct prerequisite before downstream D-REASSESS activation.

**Comment by Szczepanov** [comment]
@coderabbitai review

File: app/src/engine/intradayReassessment.ts [hunk 5]
```diff
@@ -0,0 +1,337 @@
...
+                timingPrerequisiteMet: true,
+                predecessorConfirmed: false,
+            };
+        }
+
+        // Check for acute reactive tissue responses (severe/adverse pain)
+        if (predecessor.tissueResponses) {
+            const adverseRegion = predecessor.tissueResponses.find(
+                tr => tr.afterTrainingState === 'severe' || deriveTissueSeverity(tr) === 'exclude',
+            );
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/engine/intradayReassessment.ts]: Addressed: Derived tissue severities using \deriveTissueSeverity(tr)\ and treated \severity === 'limit'\ as a scaling trigger alongside \fterTrainingState: 'mild' | 'moderate'\. Added unit test verifying that \limit\ severity from morning or post-exercise tissue responses scales PM sessions.

File: app/src/engine/intradayReassessment.ts [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+    );
+
+    if (envelopeState.mode === 'recover' && target.definition.intent !== 'recovery') {
+        return {
+            decision: 'reject',
+            reason: 'Readiness or clinical red flags require recovery mode.',
+            gateFailures: ['restricted_category'],
+            inputRevision,
+            timingPrerequisiteMet: true,
+            predecessorConfirmed: true,
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/engine/intradayReassessment.ts]: Addressed: Derived \predecessorConfirmed = target.afterSessionId !== undefined;\ and used it across all return paths, ensuring independent sessions report \predecessorConfirmed: false\ while required predecessors preserve confirmed state. Verified with unit test.

File: app/src/engine/intradayReassessment.ts [hunk 7]
```diff
@@ -0,0 +1,337 @@
...
+            : hasUnexpectedFatigue
+                ? 'Scaled authored session volume due to unexpected fatigue reported after predecessor session.'
+                : 'Scaled authored session volume due to localized muscle ache reported after predecessor session.';
+
+        return {
+            decision: 'scale',
+            reason: fatigueReason,
+            scaledDefinition: scaledDef,
+            executionDose: authoredVerdict.executionDose,
+            acceptedSystemicCost: estimateAuthoredSessionSystemicCost(scaledDef),
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/engine/intradayReassessment.ts:361]: Addressed: In the fatigue/muscle ache scaling branch, executionDose.volume is now multiplied by 0.7 while preserving intensity and other dose fields, keeping the execution dose aligned with the 0.7 scaled definition volume.

**Comment by Szczepanov** [comment]
@coderabbitai review

File: app/src/engine/intradayReassessment.test.ts [hunk 8]
```diff
@@ -0,0 +1,437 @@
...
+
+        expect(result.decision).toBe('scale');
+        expect(result.predecessorConfirmed).toBe(true);
+        expect(result.reason).toContain('unexpected fatigue');
+        expect(result.scaledDefinition).toBeDefined();
+        // Repetition sets scaled down by 0.7 (4 sets * 0.7 = 3 sets)
+        const scaledSets = result.scaledDefinition?.blocks[0].steps[0].dose;
+        if (scaledSets?.kind === 'repetition') {
+            expect(scaledSets.sets).toBe(3);
+        }
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/engine/intradayReassessment.test.ts:392]: Addressed: In \pp/src/engine/intradayReassessment.test.ts\, updated the test to assert that \scaledSets?.kind\ is \
epetition\ and unconditionally assert \expect(scaledSets).toMatchObject({ sets: 3 })\, ensuring missing or altered doses fail the test.

File: app/src/engine/intradayReassessment.ts [hunk 9]
```diff
@@ -0,0 +1,337 @@
...
+        inputRevision,
+        plannedDose,
+    } = params;
+
+    const candidateMinutes = target.definition.duration?.min ?? 45;
+    const candidateSystemicCost = estimateAuthoredSessionSystemicCost(target.definition);
+
+    // ── 1. Predecessor completion gate ─────────────────────────────────────────
+    if (target.afterSessionId) {
+        if (!predecessor || predecessor.sessionId !== target.afterSessionId || predecessor.state !== 'completed') {
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/engine/intradayReassessment.ts]: Addressed: Updated the predecessor gate in \intradayReassessment.ts\ to explicitly recognize terminal occurrence states (\superseded\, \bandoned\, and \missed\) and return a terminal \
eject\ decision with explanatory reason instead of leaving dependent sessions stuck in \pending\. Also lifted the causality check (\elapsed < 0\) outside of \minimumSeparationMinutes > 0\ so timing ordering is unconditionally checked.

**Comment by Szczepanov** [comment]
@coderabbitai review

**Comment by Szczepanov** [comment]
@coderabbitai full review

File: app/src/engine/intradayReassessment.ts [hunk 10]
```diff
@@ -0,0 +1,377 @@
...
+                reason: `Predecessor session '${target.afterSessionId}' completion timestamp is missing; cannot verify separation.`,
+                inputRevision,
+                timingPrerequisiteMet: false,
+                predecessorConfirmed: false,
+            };
+        }
+
+        // ── 2. Timing separation gate (D-TIME) ──────────────────────────────────
+        const elapsed = elapsedMinutesBetweenInstants(evaluationInstant, predecessor.completedAt);
+        if (elapsed < 0) {
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/engine/intradayReassessment.ts]: Addressed: Updated the timing separation gate in \intradayReassessment.ts\ to compare raw parsed instants (\evalMs < predMs\) and also check \Object.is(elapsed, -0)\, preventing sub-minute timestamp differences prior to predecessor completion from rounding to \-0\ and bypassing the gate. Added unit test coverage verifying that an evaluation 10 seconds before completion returns \pending\.

File: app/src/services/sessionOccurrenceService.ts [hunk 11]
```diff
@@ -134,6 +135,41 @@ export class SessionOccurrenceService {
...
+            }
+            transitioned = {
+                ...current,
+                state: 'active',
+                updatedAt: now,
+            };
+            transaction.set(ref, transitioned);
+        });
+        return transitioned!;
+    }
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/services/sessionOccurrenceService.ts:194]: Addressed: Extended \claimOccurrenceLaunch\ with \ClaimOccurrenceLaunchOptions\ and an atomic \onBeforeClaim(transaction, occurrence)\ hook executed within the Firestore transaction. This allows callers to validate \ReassessmentInputRevision\, verify date-level ledger capacity, and reserve minutes/cost atomically before transitioning the occurrence to \ctive\, aborting the transaction if validation fails. Added unit tests verifying atomic execution and abort on failure. Full UI integration in \useSessionRunner\ is sequenced in upcoming PRs (#434 PR 3 / runner orchestration).

File: docs/plans/h4-dreassess-analysis.md [hunk 12]
```diff
@@ -0,0 +1,245 @@
...
+
+2. **`reassessDependentBundleMember(...)` (pure engine function):**
+   - **Predecessor completion:** Verifies predecessor occurrence is `completed`. If not,
+     returns `pending`.
+   - **Elapsed separation:** If `minimumSeparationMinutes` is defined, computes
+     `elapsedMinutesBetweenInstants(prospectiveStartInstant, predecessorActualEndInstant)` via `localInstant.ts`.
+     If timestamps are missing or elapsed interval is insufficient, returns `pending` (unresolved timing prerequisite).
+   - **Post-predecessor confirmation:** Queries `SessionResponse` (`window: 'immediate'`)
+     and `DailySubjectiveCheckin.tissueResponses` for the predecessor. If absent, returns
+     `pending`. If adverse symptoms or reactive tissue responses are present, scales or rejects.
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply docs/plans/h4-dreassess-analysis.md]: Addressed: Updated \docs/plans/h4-dreassess-analysis.md\ to explicitly clarify that \
eassessDependentBundleMember\ is a pure engine evaluation function that evaluates the caller-supplied \SessionResponse\ and \	issueResponses\ evidence, preserving the clean separation between caller-side state reads / revision capture and pure decision evaluation.

**Comment by Szczepanov** [comment]
@coderabbitai review

File: app/src/services/sessionOccurrenceService.ts [hunk 13]
```diff
@@ -134,6 +146,49 @@ export class SessionOccurrenceService {
...
+            const parsed = parseSessionOccurrenceDocument(snap.data(), ref.path);
+            if (parsed.status !== 'AVAILABLE') {
+                throw new Error(`Occurrence ${occurrenceId} could not be parsed (${parsed.status}).`);
+            }
+            const current = parsed.data;
+            if (current.state !== 'scheduled') {
+                throw new Error(`Occurrence ${occurrenceId} cannot be claimed; state is '${current.state}', expected 'scheduled'.`);
+            }
+            if (options.onBeforeClaim) {
+                await options.onBeforeClaim(transaction, current);
```
[the remark this answers is not in the document]
Comment (Szczepanov) [reply app/src/services/sessionOccurrenceService.ts]: Addressed: In \sessionOccurrenceService.ts\, typed the occurrence passed to \onBeforeClaim\ as \Readonly<SessionOccurrence>\ and passed a defensive copy (\{ ...current, definitionRef: { ...current.definitionRef } }\). The committed active occurrence is built directly from the validated original \current\, preventing mutations in the hook from altering the stored occurrence. Added unit test verifying that mutations within \onBeforeClaim\ do not affect the committed document.

**Comment by Szczepanov** [comment]
@coderabbitai review
